### DISCUSSION TITLE
How to programatically download data from PC?

### DISCUSSION THREAD

**Discussion by jgomezdans** [microsoft/PlanetaryComputer#132] [Q&A] [answered]
I have built some Python code that creates a large bunch of files (e.g. GeoTIFFs) from processing on PC. I want to download them, which I can do by selecting them on the Jupyterlab file selection widget (which even allows a multiple files download). However, this means I have to download them to my desktop, and the on to some server where I analyse them. I'd rather just get an URL to e.g. wget or curl the data from my analysis server. Is this possible?

**Comment by TomAugspurger** [comment, marked as answer]
I'm not sure offhand. I think Jupyter / jupyterlab calls this the "contents API": https://jupyter-notebook.readthedocs.io/en/stable/extending/contents.html, but I don't know if that refers to just the UI portion, or if that also offers APIs for getting to the data.

Regardless, we recommend using Azure Blob Storage rather than your home directory for storing outputs (https://planetarycomputer.microsoft.com/docs/overview/environment/#understanding-the-file-system).
